src/Storages/StorageMergeTree.h

```diff
@@ -145,7 +145,198 @@ class StorageMergeTree final : public MergeTreeData
     /// This set have to be used with `currently_processing_in_background_mutex`.
     DataParts currently_merging_mutating_parts;
 
-    std::map<UInt64, MergeTreeMutationEntry> current_mutations_by_version;
+    class CurrentMutations
+    {
+        using MutationByVersionMap = std::map<UInt64, MergeTreeMutationEntry>;
+        using MutationReferenceByVersionMap = std::map<UInt64, std::reference_wrapper<MergeTreeMutationEntry>>;
+        using MutationReferenceByPartitionByVersionMap = std::unordered_map<String, MutationReferenceByVersionMap>;
+
+        MutationByVersionMap all_mutations;
+        MutationReferenceByPartitionByVersionMap mutations_by_partition;
```

Comment:
I think it would be better to build it dynamically (by filtering out from all_mutations items which are not related to the partition). 

Why: less RAM needed (no need to duplicate mutations for every partition), single source of truth